### PR TITLE
Remove tokio-core and replace examples with tokio

### PR DESCRIPTION
**Pull request by mehcode**

### CODE REVIEW

**Comment by lucab** [comment]
@mehcode thanks, look good! I did some cleanups to the examples in https://github.com/ctz/hyper-rustls/pull/49 but it is still not reviewd. Would you mind having a look at that and see if this PR can be easily rebased on that?

File: examples/client.rs [hunk 1]
```diff
@@ -54,8 +54,8 @@ fn main() {
             println!("Body:\n{}", String::from_utf8_lossy(&body));
         });
 
-    let mut core = tokio_core::reactor::Core::new().unwrap();
-    if let Err(err) = core.run(fut) {
+    let mut rt = tokio::runtime::Runtime::new().unwrap();
+    if let Err(err) = rt.block_on(fut) {
```
Comment (lucab) [inline examples/client.rs]: Shouldn't this be a `rt.block_on_all` (and the others in this PR too)?

**Comment by lucab** [comment]
@mehcode would you mind rebasing on current master? Also I have a minor question on executor usage.

**Comment by ctz** [comment]
BTW this is merged as-is but I'll keep this PR open to address those questions.

**Comment by lucab** [comment]
Ah dang, I missed that. Then I guess we can close https://github.com/ctz/hyper-rustls/issues/51 as well. I'm still not super familiar with the new tokio runtime, thus keeping the `block_on` vs `block_on_all` question here.

**Pull request closed without merging**